### ISSUE TITLE
[C2995]`XamlTypeInfo.g.cpp` compile error C2995 `GetReferenceTypeMember_*`  will be defined multiple times

### ISSUE THREAD

**Issue by shelllet** [microsoft/microsoft-ui-xaml#5395]
`ProjectPanel.idl` has a `Source` property.

```
namespace Velkhana
{
	[default_interface]
	runtimeclass ProjectPanel : Microsoft.UI.Xaml.Controls.Page
	{
		ProjectPanel();
		Windows.Foundation.IAsyncAction Create();
		Windows.Foundation.IAsyncAction Load(String filename);

		IObservableVector<File> Source{ get; };
	}
}

```
`XmlAttr.idl` has a same `Source` property.
```
namespace Velkhana
{
	[default_interface]
	unsealed runtimeclass XmlAttr
	{
		XmlAttr(Windows.Data.Xml.Dom.XmlElement element);
		void setXmlElement(Windows.Data.Xml.Dom.XmlElement element);
		Windows.Data.Xml.Dom.XmlDocument getXmlDocument();
		String Source;
	
		void saveXml();
		void parseXml();

		static Windows.Data.Xml.Dom.XmlDocument CreateDefault(String text, Tag tag);
	 }
}
```
`WebView2` has a `Source` property

```
 // 200 - Microsoft.UI.Xaml.Controls.WebView2.Source
    L"Source",
    &GetReferenceTypeMember_Source<::winrt::Microsoft::UI::Xaml::Controls::WebView2>,
    &SetReferenceTypeMember_Source<::winrt::Microsoft::UI::Xaml::Controls::WebView2, ::winrt::Windows::Foundation::Uri>,
    67, // Windows.Foundation.Uri
```

then compile failed:
```
1>XamlTypeInfo.g.cpp
1>C:\Users\admin\source\Velkhana\Velkhana\Velkhana\Generated Files\XamlTypeInfo.g.cpp(1294,44): error C2995: 'winrt::Windows::Foundation::IInspectable winrt::Velkhana::implementation::GetReferenceTypeMember_Source(const winrt::Windows::Foundation::IInspectable &)': function template has already been defined
1>C:\Users\admin\source\Velkhana\Velkhana\Velkhana\Generated Files\XamlTypeInfo.g.cpp(796): message : see declaration of 'winrt::Velkhana::implementation::GetReferenceTypeMember_Source'
1>C:\Users\admin\source\Velkhana\Velkhana\Velkhana\Generated Files\XamlTypeInfo.g.cpp(2966,6): error C2995: 'void winrt::Velkhana::implementation::SetReferenceTypeMember_Source(const winrt::Windows::Foundation::IInspectable &,const winrt::Windows::Foundation::IInspectable &)': function template has already been defined
1>C:\Users\admin\source\Velkhana\Velkhana\Velkhana\Generated Files\XamlTypeInfo.g.cpp(2278): message : see declaration of 'winrt::Velkhana::implementation::SetReferenceTypeMember_Source'
1>Done building project "Velkhana.vcxproj" -- FAILED.
```
`packages.config` content:
```
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="Microsoft.ProjectReunion" version="0.8.0" targetFramework="native" />
  <package id="Microsoft.ProjectReunion.DWrite" version="0.8.0" targetFramework="native" />
  <package id="Microsoft.ProjectReunion.Foundation" version="0.8.0" targetFramework="native" />
  <package id="Microsoft.ProjectReunion.InteractiveExperiences" version="0.8.0" targetFramework="native" />
  <package id="Microsoft.ProjectReunion.WinUI" version="0.8.0" targetFramework="native" />
  <package id="Microsoft.Windows.CppWinRT" version="2.0.210624.2" targetFramework="native" />
</packages>
```

**Comment by StephenLPeters** [comment]
@kennykerr @jevansaks looks like a potential cppwinrt issue, thoughts? Or is this the xaml compiler?  @alwu-msft ?

**Comment by kennykerr** [comment]
This isn't generated by cppwinrt. It's probably generated by the xaml compiler.

**Comment by alwu-msft** [comment]
@RealTommyKlein because it's suspect this is a XamlCompiler issue.

**Comment by RealTommyKlein** [comment]
Thanks for the report, this will be fixed in Reunion 1.0.  This is caused by having at least two properties with the same name (e.g. `Source`), and having at least one's property type is a non-`String` reference type while at least another's is `String`.

**Comment by RealTommyKlein** [comment]
Closing since this was fixed in Windows App SDK 1.0.
